snorkel/slicing/sf/core.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any, Callable, List, Mapping, Optional
+
+from snorkel.labeling.lf import LabelingFunction
+from snorkel.preprocess import BasePreprocessor
+
+
+class SlicingFunction(LabelingFunction):
+    """Base class for slicing functions.
+
+    See ``snorkel.labeling.lf.LabelingFunction`` for details.
+    """
+
+    pass
+
+
+class slicing_function:
```

Comment:
maybe add a quick TODO here indicating that we're planning on having a common parent decorator.